app/src/main/java/com/imdb_compose/MainActivity.kt

```diff
@@ -1,47 +1,1195 @@
+                            0 -> images.value?.backdrops!![0].file_path
+                            1 -> images.value?.backdrops!![1].file_path
+                            2 -> images.value?.backdrops!![2].file_path
+                            3 -> images.value?.backdrops!![3].file_path
+                            else -> images.value?.backdrops!![4].file_path
+                        },
+                        aspectRatio = when (i) {
+                            0 -> images.value?.backdrops!![0].aspect_ratio
+                            1 -> images.value?.backdrops!![1].aspect_ratio
+                            2 -> images.value?.backdrops!![2].aspect_ratio
+                            3 -> images.value?.backdrops!![3].aspect_ratio
+                            else -> images.value?.backdrops!![4].aspect_ratio
+                        },
+                        contentDescription = ""
+                    )
+                }
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun Carosuel(images: State<Images?>) {
+    var rowSize by remember {
+        mutableStateOf(Size.Zero)
+    }
+    val state = rememberCarouselState { 5 }
+
+    Row (
+        modifier = Modifier
+            .height(400.dp)
+            .fillMaxWidth()
+            .onGloballyPositioned { layoutCoordinates ->
+                rowSize = layoutCoordinates.size.toSize()
+            }
+    ) {
+        val width = LocalDensity.current.run { rowSize.width.toDp() }
+
+        Column {
+            HorizontalMultiBrowseCarousel(
+                state = state,
+                preferredItemWidth = width,
+                itemSpacing = 0.dp
+            ) {i ->
+                Box(modifier = Modifier.fillMaxSize()) {
+                    ImageAsync(
+                        backDropPath = when (i) {
+                            0 -> images.value?.backdrops!![0].file_path
+                            1 -> images.value?.backdrops!![1].file_path
+                            2 -> images.value?.backdrops!![2].file_path
+                            3 -> images.value?.backdrops!![3].file_path
+                            else -> images.value?.backdrops!![4].file_path
+                        },
+                        aspectRatio = when (i) {
+                            0 -> images.value?.backdrops!![0].aspect_ratio
+                            1 -> images.value?.backdrops!![1].aspect_ratio
+                            2 -> images.value?.backdrops!![2].aspect_ratio
+                            3 -> images.value?.backdrops!![3].aspect_ratio
+                            else -> images.value?.backdrops!![4].aspect_ratio
+                        },
+                        contentDescription = ""
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun isLoading() {
+    CircularProgressIndicator(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(128.dp),
```

Comment:
This seems like a big padding. If you want to limit the size of the progress indicator, you should set a max size instead of shrinking it with padding. This is probably why it looked weird when loading the details page